lib/account/pages/reviews_page.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_app/account/models/profile.dart';
+import 'package:flutter_app/rides/models/ride.dart';
+import 'package:flutter_app/util/big_button.dart';
+import 'package:flutter_app/util/profiles/profile_widget.dart';
+import 'package:flutter_app/util/profiles/reviews/aggregate_review_widget.dart';
+import 'package:flutter_app/util/supabase.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../../util/review_detail.dart';
+import '../models/review.dart';
+import 'write_review_page.dart';
+
+class ReviewsPage extends StatefulWidget {
+  final int profileId;
+  final Profile? profile;
+
+  const ReviewsPage({super.key, required this.profileId}) : profile = null;
+  ReviewsPage.fromProfile(this.profile, {super.key}) : profileId = profile!.id!;
+
+  @override
+  State<ReviewsPage> createState() => _ReviewsPageState();
+}
+
+class _ReviewsPageState extends State<ReviewsPage> {
+  late final int _profileId;
+  Profile? _profile;
+  bool _reviewable = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _profileId = widget.profileId;
+      if (widget.profile != null) _profile = widget.profile!;
+    });
+
+    load();
+  }
+
+  Future<void> load() async {
+    Map<String, dynamic> profileData = await supabaseClient.from('profiles').select('''
+      *,
+      reviews_received: reviews!reviews_receiver_id_fkey(
+        *,
+        writer: writer_id(*)
+      )''').eq('id', _profileId).single();
+    List<dynamic> commonRidesData = await supabaseClient.from('rides').select('''
+          *,
+          drive: drives!inner(*)
+        ''').eq('rider_id', SupabaseManager.getCurrentProfile()!.id).eq('drive.driver_id', _profileId);
+
+    setState(() {
+      _profile = Profile.fromJson(profileData);
+      List<Ride> rides = Ride.fromJsonList(commonRidesData);
+      _reviewable = rides.any((ride) => ride.isFinished && ride.status == RideStatus.approved);
```

Comment:
this should also check if there is an existing review and change the button then?

Reply:
that would probably be a good idea otherwise that might be confusing if I'm riding with a person again and don't remember...

Reply:
Done.